src/gui/render/render.cpp

```diff
@@ -542,13 +542,28 @@ bool RENDER_MaybeAutoSwitchShader(const DosBox::Rect canvas_size_px,
                                   const VideoMode& video_mode,
                                   const bool reinit_render)
 {
+	const auto curr_preset = GFX_GetRenderer()->GetCurrentShaderPreset(); //-V821
+
 	const auto shader_changed = GFX_GetRenderer()->MaybeAutoSwitchShader(
 	        canvas_size_px, video_mode);
 
+	const auto new_preset = GFX_GetRenderer()->GetCurrentShaderPreset(); //-V821
+
```

Comment:
Two more deep copies of these presets. `new_preset` should at least be inside the `if (shader_changed)` block.

Reply:
Move it inside the `if`.